qt-models/datestatsmodel.cpp

```diff
@@ -0,0 +1,141 @@
+#include "datestatsmodel.h"
+#include "core/dive.h"
+#include "core/divelist.h"
+#include "core/time.c"
+#include <QtCore/QVector>
+#include <QtCore/QRandomGenerator>
+#include <QtCore/QRect>
+#include <QtGui/QColor>
+
+DateStatsTableModel::DateStatsTableModel(QObject *parent) :
+        QAbstractTableModel(parent)
+{
+        int idx;
+        struct dive *dp;
+        struct tm tm;
+        m_rowCount = YEAR;
+        int current_year = -1;
+        QVector<int>* yearVec;
```

Comment:
No pointer here, please.

Reply:
If I'm interpreting the code right, I would initialize this by a fixed size vector
```
QVector<int> yearVec(YEAR + 1, 0);
```
Then add the vector with
```
m_data.append(yearVec);
```
and reset it with
```
yearVec.fill(0);
yearVec[YEAR] = xxx;
```